Plugins/RenderTargetSerializerPlugin/Source/RenderTargetSerializer/Private/RenderTargetSerializerBPLibrary.cpp

```diff
@@ -36,53 +40,67 @@ TArray<FVector> URenderTargetSerializerBPLibrary::SerializeRenderTarget(UTexture
     TArray<FColor> PixelData;
     PixelData.Init(FColor::Black, Width * Height);
 
+    Channels.Empty(Width * Height * 3);
+
     //// Read pixel data from the render target
     FRenderTarget* RenderTargetResource = RenderTarget->GameThread_GetRenderTargetResource();
     FReadSurfaceDataFlags ReadPixelFlags;
-    ReadPixelFlags.SetLinearToGamma(false); // Ensure no colour space conversion
+    // Ensure no colour space conversion (do we want this? for uint8 we *do* want gamma space.)
+    ReadPixelFlags.SetLinearToGamma(false);
     RenderTargetResource->ReadPixels(PixelData, ReadPixelFlags);
 
     for (int32 y = 0; y < Height; ++y)
     {
         for (int32 x = 0; x < Width; ++x)
         {
-            // Convert pixel colour to vector format (XYZ = RGB)
+            // Convert rgba32f pixel colour to rgb8ui format
             FColor PixelColor = PixelData[y * Width + x];
-            FVector PixelVector(PixelColor.R, PixelColor.G, PixelColor.B);
 
-            // Add pixel vector to the array
-            PixelVectors.Add(PixelVector);
+            uint8 Rgb[3] = {
+                PixelColor.R,
+                PixelColor.G,
+                PixelColor.B
+            };
+
+            // Add three channels into the vector
+            Channels.Append(Rgb, sizeof(Rgb) / sizeof(Rgb[0]));
         }
     }
 
-    return PixelVectors; //SerializedData;
+    return Channels;
 }
 
-UTexture2D* URenderTargetSerializerBPLibrary::DeserializeRenderTarget(const TArray<FVector>& PixelVectors, int32 Width, int32 Height)
+UTexture2D* URenderTargetSerializerBPLibrary::DeserializeRenderTarget(const TArray<uint8>& Channels, int32 Width, int32 Height)
 {
-    if (Width <= 0 || Height <= 0 || PixelVectors.Num() != Width * Height)
+    if (Width <= 0 || Height <= 0 || Channels.Num() != Width * Height * 3)
     {
         return nullptr;
     }
 
-    // Create a texture 2D to store the pixel data
+    // Create a texture 2D to store the pixel data (there is no RGB format! weird)
     UTexture2D* Texture2D = UTexture2D::CreateTransient(Width, Height, PF_B8G8R8A8);
     if (!Texture2D)
     {
         return nullptr;
     }
 
     // Lock the texture for writing
-    FTexture2DMipMap& Mip = Texture2D->PlatformData->Mips[0];
+    FTexture2DMipMap& Mip = Texture2D->GetPlatformData()->Mips[0];
```

Comment:
GetPlatformData is not available in Texture2D for Unreal 4.27 or I'm missing something?